x-pack/legacy/plugins/epm/server/lib/kibana/index_pattern/install.test.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import path from 'path';
+import { readFileSync } from 'fs';
+import { safeLoad } from 'js-yaml';
+import { flattenFields, dedupFields } from './install';
+import { Field } from '../../fields/field';
+
+// Add our own serialiser to just do JSON.stringify
+expect.addSnapshotSerializer({
+  print(val) {
+    return JSON.stringify(val, null, 2);
+  },
+
+  test(val) {
+    return val;
+  },
+});
+
```

Comment:
Could we change this to confirm that `makeKibanaIndexPatternFields` properly transforms `Field[]` to `IndexPatternField[]` instead of testing the internals of that function?

i.e. a test to confirm `const kibanaIndexPatternFields = makeKibanaIndexPatternFields(allFields);` works as expected; not that `makeKibanaIndexPatternFields` calls some specific functions.

Reply:
this is something i'm going to deal with in the next PR.  I haven't figured it out yet but I think I might be getting rid of IndexPatternField[] and just using Field[].  I created the former temporarily because Field[] wasn't working for me but I should be able to just modify Field[].  I need to figure out whether I can use it for both fields for mappings and kibana index patterns.